.github/workflows/server-cargo-clean.yaml

```diff
@@ -0,0 +1,40 @@
+on: 
+  pull_request: 
+    branches: 
+      - main
+      - develop
+    paths: 
+      - 'server/**'
```

Comment:
Awesome! Yeah let's remove this piece now that CI is passing, once we merge into develop (our default branch) the workflow_dispatch should be available :)